src/Microsoft.AspNetCore.Routing/Matchers/DictionaryJumpTable.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class DictionaryJumpTable : JumpTable
+    {
+        private readonly int _default;
+        private readonly int _exit;
+        private readonly Dictionary<string, int> _dictionary;
+
+        public DictionaryJumpTable(int @default, int exit, (string text, int destination)[] entries)
+        {
+            _default = @default;
+            _exit = exit;
+
+            _dictionary = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+            for (var i = 0; i < entries.Length; i++)
+            {
+                _dictionary.Add(entries[i].text, entries[i].destination);
+            }
+        }
+
+        public override int GetDestination(string path, PathSegment segment)
+        {
+            if (segment.Length == 0)
+            {
+                return _exit;
+            }
+
+            var text = path.Substring(segment.Start, segment.Length);
```

Comment:
Yeah 😢  - this is a longstanding problem in routing. We need case-insensitive hashing of portions of strings.

@GrabYourPitchforks is going to help us with this, and I'd prefer to use whatever he's baking rather than some of the esoteric things I developed in #591

Reply:
Either way I expect this code as-is to be temporary, but at least it's working and simple and temporary.

Reply:
What about a struct key that is a string+index+length and implements GetHashCode?

Reply:
Ohh, is the issue is the case-insensitive hash from the segment?

Reply:
Yep.

Reply:
So there are things that are more esoteric we can do to avoid the substring. But help is on the way from ya boy, so I figured we should let them handle that.